docs: Add docs for PKCS#11 provider (#17312)
hashicorp/vault: website/content/docs/enterprise/pkcs11-provider/oracle-tde.mdx

---
layout: docs
page_title: Oracle TDE - PKCS#11 Provider - Vault Enterprise
description: |-

>   Vault Enterprise ...

Reviewer: Needs more description here?

---

# Oracle TDE

> [Oracle Transparenet Data Encryption](https://docs.oracle.com/database/121/ASOAG/introduction-to-transparent-data-encryption.htm#ASOAG10270) (TDE)

Reviewer: ```suggestion
[Oracle Transparent Data Encryption](https://docs.oracle.com/database/121/ASOAG/introduction-to-transparent-data-encryption.htm#ASOAG10270) (TDE)
```

is supported with the [Vault PKCS#11 provider](/docs/enterprise/pkcs11-provider).

> In this setup, Vault's KMIP engine will generate and store the "TDE Master Encryption Key" that the Oracle Database will use to encrypt and decrypt the "TDE Table Keys".

Reviewer 2: ```suggestion
In this setup, Vault's KMIP engine generates and store the "TDE Master Encryption Key" that the Oracle Database uses to encrypt and decrypt the "TDE Table Keys".
```

---
layout: docs
page_title: Oracle TDE - PKCS#11 Provider - Vault Enterprise
description: |-
  The Vault PKCS#11 Provider can be used to enable Oracle TDE.
---

# Oracle TDE

[Oracle Transparent Data Encryption](https://docs.oracle.com/database/121/ASOAG/introduction-to-transparent-data-encryption.htm#ASOAG10270) (TDE)
is supported with the [Vault PKCS#11 provider](/docs/enterprise/pkcs11-provider).
In this setup, Vault's KMIP engine generates and store the "TDE Master Encryption Key" that the Oracle Database uses to encrypt and decrypt the "TDE Table Keys".
Oracle will not have access to the TDE Master Encryption Key itself.

## Requirements

To setup Oracle TDE backed by Vault, the following are required:

- A database running Oracle 19 Enterprise Edition
- A Vault Enterprise 1.11+ server with Advanced Data Protection for KMIP support.
- Vault has TCP port 5696 accessible to the Oracle database.
- `libvault-pkcs11.so` downloaded from [releases.hashicorp.com](https://releases.hashicorp.com/vault-pkcs11-provider) for the operating system running the Oracle database (the RHEL 7 x86-64 version for Oracle Enterprise Linux 7).

## Vault Setup

> On the Vault server, we need to [setup the KMIP Secrets Engine](https://www.vaultproject.io/docs/secrets/kmip), e.g.:

Reviewer: ```suggestion
On the Vault server, we need to [setup the KMIP Secrets Engine](/docs/secrets/kmip), e.g.:
```

1. Start the KMIP Secrets Engine and listener:

    ```sh
    vault secrets enable kmip
    vault write kmip/config listen_addrs=0.0.0.0:5696
    ```

    ~> **Important**: When configuring KMIP for Oracle, you will probably need to set the

>     `server_hostnames` and `server_ips` [configuration parameters](https://www.vaultproject.io/api-docs/secret/kmip#parameters),

Reviewer: ```suggestion
    `server_hostnames` and `server_ips` [configuration parameters](/api-docs/secret/kmip#parameters),
```
